wagtail/contrib/modeladmin/views.py

```diff
@@ -697,6 +730,7 @@ def check_action_permitted(self, user):
 
     @method_decorator(login_required)
     def dispatch(self, request, *args, **kwargs):
+        self.instance
```

Comment:
This could do with a comment to explain why it's here, something like: "pre-fetch the instance to ensure that any Http404 exception happens before the redirect below"